File: CivitasJ/src/civitas/Dado.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.Random;

/**
 *
 * @author ramonjtorres
 */
class Dado {
  
  private Random random;
  private int ultimoResultado;
  private boolean debug;
  
  static final private Dado instance = new Dado();
  static private int SalidaCarcel = 5; 
  
  private Dado() {
    
      random = new Random();
      ultimoResultado = 0;
      debug = false;
  }
  
  static Dado getInstance() {
    return instance;
  }
  
  int tirar(){
      
      if(!debug){
          
          ultimoResultado = random.nextInt(6)+1;
      }
      else{
          
          ultimoResultado = 1;
      }
      
      return ultimoResultado;
  }
  
  boolean salgoDeLaCarcel(){
  
      if(tirar() >= SalidaCarcel){ //falta cambiarlo en ruby tambien para que sera mayor o igual
      
          return true;
      }
      
      return false;
  }
  
  int quienEmpieza(int n){
  
      return random.nextInt(n);
  }
  
  void setDebug(boolean d){
  
      debug = d;
      Diario.getInstance().ocurreEvento("Modo debug activado");
  }
  
   int getUltimoResultado(){
  
      return ultimoResultado;
  }

}

File: CivitasR/lib/dado.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require 'singleton'

module Civitas
  public class Dado
    include Singleton
    
    attr_reader :ultimo_Resultado
    attr_writter :debug
    
    @salida_carcel = 5
    
    def initialize
      
      @random = rand()
      @ultimo_resultado = 0
      @debug = false
      
    end
    
    public def tirar()
      
      if(@debug)
       
        @ultimo_resultado = rand(5) + 1
        
      else
      
        @ultimo_resultado = 1
      
      end
      
      return @ultimo_resultado
      
    end
    
    public def salgo_de_la_carcel()
      
      if(tirar() == @salida_carcel)
        
        return true
        
      end
      
      return false
      
    end
    
    public def quien_empieza(n)
      
      return rand(n-1)
      
    end
  end
end
